src/vm/fieldmarshaler.h

```diff
@@ -84,13 +84,11 @@ enum NStructFieldType
 
 //=======================================================================
 // Magic number for default struct packing size.
+//
+// Currently we set this to the packing size of the largest supported
+// fundamental type and let the field marshaller downsize where needed.
 //=======================================================================
-#if defined(_TARGET_X86_) && defined(UNIX_X86_ABI)
-// A double is 4-byte aligned on GCC (without -malign-dobule)
-#define DEFAULT_PACKING_SIZE 4
-#else // _TARGET_X86_ && UNIX_X86_ABI
-#define DEFAULT_PACKING_SIZE 8
-#endif // !_TARGET_X86_ || !UNIX_X86_ABI
+#define DEFAULT_PACKING_SIZE 32
```

Comment:
This is undoing change https://github.com/dotnet/coreclr/pull/10340 . How to think about that?

Reply:
The second commit changes it so that packing/alignment is handled in methodtable builder (as is done with special packing/alignments for all other types): https://github.com/dotnet/coreclr/pull/15961/commits/850eda4d4db15c6f005ddf0d944142465728307e#diff-f2c1beb35334bda2c2e3c3cbbc0ccd54R9586

Reply:
There is also an explicit test covering that scenario here: https://github.com/dotnet/coreclr/pull/15961/files#diff-d13baf2c92ef528b9520b16eec06fdeaR290 (and the same for `ulong`, and `long`)